platforms/jvm/code-quality/src/main/java/org/gradle/api/plugins/quality/PmdExtension.java

```diff
@@ -216,22 +162,18 @@ public void setRuleSetFiles(FileCollection ruleSetFiles) {
      * @param ruleSetFiles the rule set files to be added
      */
     public void ruleSetFiles(Object... ruleSetFiles) {
-        this.ruleSetFiles.from(ruleSetFiles);
+        getRuleSetFiles().from(ruleSetFiles);
     }
 
     /**
      * Whether or not to write PMD results to {@code System.out}.
      */
-    @ToBeReplacedByLazyProperty
-    public boolean isConsoleOutput() {
-        return consoleOutput;
-    }
+    @ReplacesEagerProperty(originalType = boolean.class)
+    public abstract Property<Boolean> getConsoleOutput();
 
-    /**
-     * Whether or not to write PMD results to {@code System.out}.
-     */
-    public void setConsoleOutput(boolean consoleOutput) {
-        this.consoleOutput = consoleOutput;
+    public Property<Boolean> getIsConsoleOutput() {
+        // TODO: Deprecate
+        return getConsoleOutput();
```

Comment:
:x: This should be marked `@Internal`. I wonder why we don't get a build failure because of this; we are supposed to check all the task instances... ah, but this is not a task instance, but an extension, that's why!